File: is52038b-labs/06/java/StackQueue.java
public class StackQueue {
    private static Stack s;
    private static Queue q;
    public static void main(String[] args) {
	prepare();
	System.out.print(q.dequeue());
	System.out.print(s.pop());
	System.out.print(s.pop());
	System.out.print(q.dequeue());
	s.pop();
	q.dequeue();
	System.out.print(s.pop());
	System.out.print(q.dequeue());
	System.out.print(s.pop());
	System.out.println(q.dequeue());
	s.pop();
	q.dequeue();
    }
    private static void prepare() {
    }
}

File: is52038b-labs/06/cpp/StackQueue.cpp
#include <iostream>

#include "Stack.hpp"
#include "Queue.hpp"

using namespace std;

static Stack *s;
static Queue *q;

void prepare() {
}

int main(int argc, char *argv[]) {
  prepare();
  cout << q->dequeue();
  cout << s->pop();
  cout << s->pop();
  cout << q->dequeue();
  s->pop();
  q->dequeue();
  cout << s->pop();
  cout << q->dequeue();
  cout << s->pop();
  cout << q->dequeue() << endl;
  s->pop();
  q->dequeue();
}
